quinn-udp/src/unix.rs

```diff
@@ -794,21 +794,120 @@ pub(crate) const BATCH_SIZE: usize = 32;
 #[cfg(apple_slow)]
 pub(crate) const BATCH_SIZE: usize = 1;
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod linux {
+    pub(crate) fn kernel_version_string() -> Result<String, libc::c_int> {
+        let mut n = unsafe { std::mem::zeroed() };
+        let r = unsafe { libc::uname(&mut n) };
+        if r != 0 {
+            return Err(r);
+        }
+        Ok(unsafe {
+            std::ffi::CStr::from_ptr(n.release[..].as_ptr())
+                .to_string_lossy()
+                .into_owned()
+        })
+    }
+
+    // https://www.linfo.org/kernel_version_numbering.html
+    #[derive(Eq, PartialEq, Ord, PartialOrd, Debug)]
+    pub(crate) struct KernelVersion {
+        version: u8,
+        major_revision: u8,
+    }
+
+    impl KernelVersion {
+        pub(crate) const fn new(version: u8, major_revision: u8) -> Self {
+            Self {
+                version,
+                major_revision,
+            }
+        }
+
+        pub(crate) fn from_str(release: &str) -> Result<Self, String> {
+            let mut split = release
+                .split_once('-')
+                .map(|pair| pair.0)
+                .unwrap_or(release)
+                .split('.');
+
+            let version = split
+                .next()
+                .and_then(|s| s.parse().ok())
```

Comment:
Nit: please use `u8::from_str()` instead of `.parse()`, and avoid capitalizing error messages.

Reply:
> use u8::from_str() instead of .parse()

This will require `use std::str::FromStr;`, are you ok with it?

Reply:
Yes; that's the same stuff that `parse()` uses.

Reply:
> avoid capitalizing error messages

Also for "GSO" in "GSO disabled:..."?

Reply:
Sorry, no -- just the first letter.